### PR TITLE
Add metadata to firefox schemas

### LINKED ISSUE #1130: Add missing metadata to mozilla-central manifest schemas
The manifest schema is being imported from mozilla-central in #1129. mozilla-central doesn't have all of the data that we need though, so we will need to add some of our metadata to the extracted schemas.

Things that are missing are some descriptions and validations such as the `strict_min_version` which mozilla-central just marks as optional but we have a regex to validate. For the `strict_min_version` case there is likely some validation being done elsewhere in Firefox. We should see about matching Firefox's validation and moving that into the schema, even if only as a `format`.

Depends on #1129.

### PR DESCRIPTION
**Pull request by mstriemer**
Update the imported schemas so that they validate everything that we had in our unit tests. I'm still going to go through and test this some more manually to ensure that the output is about the same. I know some things now give multiple errors instead of just one so I'm not sure how that will format.

This also updates the validator to support reading the schema data from the firefox schemas or the linter, this is controlled by the `USE_FIREFOX_SCHEMAS` environment variable. The tests pass under both versions and they will both be run on Travis.

Fixes #1130.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +52,7 @@
     "json-loader": "0.5.4",
     "load-grunt-configs": "1.0.0",
     "load-grunt-tasks": "3.5.2",
+    "lodash": "^4.17.4",
```
Comment (pdehaan) [inline package.json]: Do you need all of **lodash**, or just [**lodash.ismatchwith**](https://www.npmjs.com/package/lodash.ismatchwith)?

Comment (mstriemer) [inline package.json]: That should work. I wasn't too concerned about it since this is just for the tests but may as well be specific about it.

**Comment by tofumatt** [comment]
This is a big diff so I'm happy to review but there is one more mobile v1 bug I'd like to get out first so I won't be able to look at this until the weekend/Monday. Hope that's okay! 😄

**Comment by mstriemer** [comment]
Lots of changes but I think most of them are the schema files and test.manifestjson.js where I move all the top-level describes into a main describe block with the shared prefix they all had. Might want to check that file with [whitespace ignored](https://github.com/mozilla/addons-linter/pull/1169/files?w=1).

**Comment by tofumatt** [comment]
Yeah fair enough. Also OMG that is a cool trick ^_^

**Comment by mstriemer** [comment]
Yeah, I was going to assign it to you when I first submitted this but figured you were probably busy 😛. No rush!

**Comment by mstriemer** [comment]
@cgrebs took a look at #1175 which includes these changes, I made some updates here for his comments. Merging based on that review.

**Pull request merged**